src/core/exception.d

```diff
@@ -47,29 +47,121 @@ else:
  */
 class RangeError : Error
 {
-    this( string file = __FILE__, size_t line = __LINE__, Throwable next = null ) @nogc nothrow pure @safe
+    /// lower/upper bound passed, if slice (array[lower .. upper]), if not a slice (array[upper])
+    const size_t lower, upper;
+    /// length of indexed array
+    const size_t length;
+
+    this( string file = __FILE__, size_t line = __LINE__, Throwable next = null) @nogc nothrow pure @safe
+    {
+        this( 0, 0, 0, file, line, next);
+    }
+
+    this( size_t lower, size_t upper, size_t length, string file = __FILE__,
+          size_t line = __LINE__, Throwable next = null) @nogc nothrow pure @safe
     {
+        this.lower  = lower;
+        this.upper  = upper;
+        this.length = length;
         super( "Range violation", file, line, next );
     }
+
+    override void toString(scope void delegate(in char[]) sink) const
+    {
+        import core.internal.string : unsignedToTempString;
+
+        char[20] tmpBuff = void;
+
+        sink(typeid(this).name);
+        sink("@"); sink(file);
+        sink("("); sink(unsignedToTempString(line, tmpBuff, 10)); sink(")");
+
+        sink(": "); sink(msg);
+        if (upper)
+        {
+            sink(lower ? ": indicies [" : ": index [");
+            if (lower)
+            {
+                sink(unsignedToTempString(lower, tmpBuff, 10)); sink(" .. ");
+            }
+            sink(unsignedToTempString(upper, tmpBuff, 10));
+            sink("] exceeds array length ");
+            sink(unsignedToTempString(length, tmpBuff, 10));
+        }
+
+        if (info)
+        {
+            try
+            {
+                sink("\n----------------");
+                foreach (t; info)
+                {
+                    sink("\n"); sink(t);
+                }
+            }
+            catch (Throwable)
+            {
+                // ignore more errors
+            }
+        }
+    }
 }
 
 unittest
 {
     {
-        auto re = new RangeError();
+        auto re = new RangeError(0, 42, 7);
         assert(re.file == __FILE__);
         assert(re.line == __LINE__ - 2);
         assert(re.next is null);
         assert(re.msg == "Range violation");
     }
 
     {
-        auto re = new RangeError("hello", 42, new Exception("It's an Exception!"));
+        auto re = new RangeError(0, 42, 7, "hello", 42, new Exception("It's an Exception!"));
         assert(re.file == "hello");
         assert(re.line == 42);
         assert(re.next !is null);
         assert(re.msg == "Range violation");
     }
+
+    {
+        int[] a = [1, 2, 3];
+        //try
+        //{
+            // This is necessary becase the unittests are compiled with -release
```

Comment:
Nope, we have have `unittest-debug` and `unittest-release` targets (both are run when you run `unittest`). Simply version these test out for release builds.